desc/objectives/_coils.py

```diff
@@ -607,6 +607,128 @@ def compute(self, params, constants=None):
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+        be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Operates over all coils, not each individual coil.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+        Defaults to ``LinearGrid(N=2 * coil.N + 5)``
+    name : str, optional
+        Name of the objective function.
+    """
+
+    _scalar = False
+    _units = "(A*m)"
+    _print_value_fmt = "Coil current length: {:10.3e} "
+
+    def __init__(
+        self,
+        coil,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name="coil current length",
+    ):
+        if target is None and bounds is None:
+            target = 0
+
+        super().__init__(
+            coil,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            grid=grid,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        super().build(use_jit=use_jit, verbose=verbose)
+
+        if self._normalize:
+            coils = self.things[0]
+            params = tree_flatten(
+                coils.params_dict, is_leaf=lambda x: isinstance(x, dict)
+            )[0]
+            average_current = np.mean([param["current"] for param in params])
+            if not average_current:
+                average_current = 1
+            self._normalization = 1 / (average_current * self._scales["a"])
```

Comment:
The normalization should be the inverse of this (current * a) so that it has the same units as the objective compute values (A*m). The objective residuals are divided by the normalization, not multiplied.

Reply:
I think this and the previous comment have been resolved?